src/main/java/org/jabref/gui/util/DefaultTaskExecutor.java

```diff
@@ -23,21 +23,35 @@
 
     private static final Log LOGGER = LogFactory.getLog(DefaultTaskExecutor.class);
 
-    private ExecutorService executor = Executors.newFixedThreadPool(5);
+    private final ExecutorService executor = Executors.newFixedThreadPool(5);
 
     public static <V> V runInJavaFXThread(Callable<V> callable) {
-        FutureTask<V> task = new FutureTask<>(callable);
-        Platform.runLater(task);
-        try {
-            return task.get();
-        } catch (InterruptedException | ExecutionException e) {
-            LOGGER.error(e);
-            return null;
+        if (!Platform.isFxApplicationThread()) {
+            FutureTask<V> task = new FutureTask<>(callable);
+            Platform.runLater(task);
+            try {
+                return task.get();
+            } catch (InterruptedException | ExecutionException e) {
+                LOGGER.error(e);
+                return null;
+            }
+        } else {
+            try {
+                //we directly execute the callback
+                return callable.call();
```

Comment:
runLater is async, while this code is sync. I don't like that we change behavior depending on whether we are in a javafx thread or not. What was your idea behind those changes?

Reply:
I did not really think about it. will wrap it.